DALK.PL_ANALYZER/Models/Matches/MatchesModelViewFactory.cs

```diff
@@ -13,35 +13,52 @@ public class MatchesModelViewFactory
     {
         private FakeDB db;
         public MatchesModelView matchesMV;
-        public MatchesModelViewFactory(int? matchSeasonsId = null, int? matchLeaguesId = null, int? matchTeamsId = null, int? matchGroupId = null, string matchStagesId = null)
+        public MatchesModelViewFactory(RawFilterValues rawParameters)
         {
+            List<GridFilter> allFilters = new List<GridFilter>();
+            MatchesFiltersValues filterValues = new MatchesFiltersValues(rawParameters);
             db = new FakeDB();
 
-            LeaguesSeason leaguesSeason = db.GetLeaguesSeason();
-            TeamsSeason teamSeason = db.GetTeamsSeason();
             Seasons seasons = db.GetSeasons();
             SeasonFilterData defaultSeason = seasons.GetDefaultSeason();
+            GridFilterFactory factory = new GridFilterFactory(seasons.GetSeasons().ToList<IFilterData>(), "Każdy sezon", "matchSeason");
+            factory.SetDefaultFilter(defaultSeason);
+            factory.SetEmptyFilterNotSelected();
+            factory.SetIdByDefault(rawParameters.matchSeasonsId);
+            allFilters.Add(factory.GetGridFilter());
```

Comment:
Here, use it like this:
GetFilter(`defaultSeson, rawParameters.matchSeasonsId`)